tests/events/events.go

```diff
@@ -0,0 +1,107 @@
+package events
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests/util"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// ExpectNoEvent is safe to use in parallel as long as you are asserting namespaced object that is not shared between tests
+func ExpectNoEvent(object any, eventType, reason string) {
```

Comment:
I didn't know `any` existed until now :)

- What is the reason for using `any` instead of `interface{}`?
- This seems to break type checking, and causes you to embed deeply assertions. Isn't it possible to just use `runtime.Object` or `schema.ObjectKind` as the required type/interface?

Reply:
`any` is just good old `interface{}`. Both interfaces would not help me or am I missing something? (well I could remove one assertion)

Reply:
I think `any` was introduced to read better when defining generics, but I am not really sure.
It is however confusing to see them interchangeably in the code base, when meaning the same (as regular interface in this case).

> Both interfaces would not help me or am I missing something? (well I could remove one assertion)

I got a bit lost in all the conversions there, now I see you use the result of multiple conversions.
I would suggest trying to use "k8s.io/apimachinery/pkg/api/meta" which should provide access to generic fields like these ones.